dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java

```diff
@@ -1897,7 +1897,7 @@ public void submitStandByTask() throws StateEventHandleException {
         TaskInstance task;
         while ((task = readyToSubmitTaskQueue.peek()) != null) {
             // stop tasks which is retrying if forced success happens
-            if (task.taskCanRetry()) {
+            if (task.getId() != null && task.taskCanRetry()) {
```

Comment:
In which case, the taskId will be null but it can retry?

Reply:
When a task fails and retried, a new task instance will be created using the retryTaskInstance method, and then inserted into the readyToSubmitTaskQueue using the addTaskToStandByList method. The instance ID is null and the status is failed